app/src/main/java/org/stepik/android/domain/course_calendar/interactor/CourseCalendarInteractor.kt

```diff
@@ -0,0 +1,63 @@
+package org.stepik.android.domain.course_calendar.interactor
+
+import android.content.Context
+import io.reactivex.Completable
+import io.reactivex.Single
+import io.reactivex.rxkotlin.toObservable
+import org.stepic.droid.R
+import org.stepic.droid.model.CalendarItem
+import org.stepic.droid.util.doCompletableOnSuccess
+import org.stepik.android.domain.calendar.model.CalendarEventData
+import org.stepik.android.domain.calendar.repository.CalendarRepository
+import org.stepik.android.domain.course_calendar.model.SectionDateEvent
+import org.stepik.android.domain.course_calendar.repository.CourseCalendarRepository
+import org.stepik.android.view.course_content.model.CourseContentItem
+import javax.inject.Inject
+
+class CourseCalendarInteractor
+@Inject
+constructor(
+    private val context: Context,
+    private val calendarRepository: CalendarRepository,
+    private val courseCalendarRepository: CourseCalendarRepository
+) {
+    fun getCalendarItems(): Single<List<CalendarItem>> =
+            calendarRepository.getCalendarItems()
+
+    fun applyDatesToCalendar(dates: List<CourseContentItem>, calendarItem: CalendarItem): Completable {
```

Comment:
It will be better to rename this method and its arguments to something more meaningful. Like export schedule to calendar.